src/Microsoft.ComponentDetection.Common/FileUtilityService.cs

```diff
@@ -36,4 +37,26 @@ public Stream MakeFileStream(string fileName)
     {
         return new FileStream(fileName, FileMode.Open, FileAccess.Read);
     }
+
+    /// <inheritdoc />
+    public (string DuplicateFilePath, bool CreatedDuplicate) DuplicateFileWithoutLines(string fileName, string removalIndicator)
+    {
+        // Read all lines from the file and filter out the lines that start with the removal indicator.
+        var originalLines = File.ReadLines(fileName).ToList();
```

Comment:
nit: Do we need to enumerate files before loop below, we would be effectively listing items twice, not a big concern tho, just more of an FYI

Reply:
Hmm I think with Linq we need to loop twice, since we need to count the orignal enumerable as well as the new one. Let me update to a single loop